### PR TITLE
Squash o.e.rest.action.admin.indices

### PR DESCRIPTION
**Pull request by nik9000**
In an effort to reduce the number of tiny packages we have in the
code base this moves all the files that were in subdirectories of
`org.elasticsearch.rest.action.admin.indices` into
`org.elasticsearch.rest.action.admin.indices`.

Also fixes line length in these packages. It makes a single non-checkstyle change: implementing `ToXContent` on `GetIndexTemplatesResponse`. I did this because it was the right thing to do and it fixed a line length violation.

### CODE REVIEW

**Comment by nik9000** [comment]
Whoever draws the short straw to review this probably should do it commit by commit.

File: buildSrc/src/main/resources/checkstyle_suppressions.xml [hunk 1]
```diff
@@ -507,26 +507,12 @@
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]repositories[/\\]uri[/\\]URLRepository.java" checks="LineLength" />
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]rest[/\\]BytesRestResponse.java" checks="LineLength" />
   <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]rest[/\\]RestController.java" checks="LineLength" />
-  <suppress files="core[/\\]src[/\\]main[/\\]java[/\\]org[/\\]elasticsearch[/\\]rest[/\\]action[/\\]admin[/\\]cluster[/\\]health[/\\]RestClusterHealthAction.java" checks="LineLength" />
```
Comment (nik9000) [inline buildSrc/src/main/resources/checkstyle_suppressions.xml]: Actually I'm not sure why this line is there. Maybe it snuck in. Checking.

Comment (nik9000) [inline buildSrc/src/main/resources/checkstyle_suppressions.xml]: It did, but I fixed the checkstyle issue anyway so I'm keeping it.

**Comment by nik9000** [comment]
> Whoever draws the short straw to review this probably should do it commit by commit.

Actually it doesn't look too bad all together.

**Comment by nik9000** [comment]
I did a quick count and this removes 31 packages.

**Comment by rjernst** [comment]
lgtm

**Comment by nik9000** [comment]
Thanks for reviewing @rjernst!

**Comment by javanna** [comment]
I think this one should be marked breaking? Java api is affected by these changes no?

**Comment by nik9000** [comment]
I'm fairly sure it is just REST moving around so it isn't breaking any transport client users but I'll mark it that way anyway just in case.

**Comment by javanna** [comment]
you don't have to mark it breaking if it's not :) I double checked and you are right, I thought you had moved also response and request classes. sorry for the confusion.

**Comment by nik9000** [comment]
> you don't have to mark it breaking if it's not :) I double checked and you are right, I thought you had moved also response and request classes. sorry for the confusion.

Yeah, I didn't move those classes. Squashing the REST stuff is fairly obviously right. Squashing the request/response/action/transport action/builder groups is less obvious because 5 classes is a fairly reasonable number to have in a package. Low, but defensible.

**Comment by rjernst** [comment]
>  5 classes is a fairly reasonable number to have in a package.

I don't think it is defensible! What does having a package for those 5 classes solve? Vs all of the request/response/action for many requests being in the same package (say all admin actions)?

**Comment by nik9000** [comment]
> I don't think it is defensible! What does having a package for those 5 classes solve? Vs all of the request/response/action for many requests being in the same package (say all admin actions)?

Yeah. I mean, I wouldn't do it if it were my project. I've certainly contributed to the layout because it looked like how everything else was done.

I think **all** admin actions might be too much. This just squashed the rest for indices admin into one package and it is a health size. Five times larger is more than I'd like.

I'm taking the cowards way and doing the uncontroversial changes to reduce packages. One class in a package is usually silly.

**Pull request merged**